source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -1325,6 +1426,13 @@ ServerContextImpl::selectTlsContext(const SSL_CLIENT_HELLO* ssl_client_hello) {
       break;
```

Comment:
Hm, so let's say the TLS config is the following, in order:
1. non-ECDSA cert with valid OCSP response
2. ECDSA cert with invalid OCSP response

For a ECDSA-capable client,  it would currently select the second and fail because `selected_ctx` does not have a valid OCSP response.

If instead we make the loop condition `client_ecdsa_capable == ctx.is_ecdsa_ && passesOcspPolicy(...)`, then we would fall back on the first context. Since the first context has a valid OCSP response we would _not_ fail the connection.

Either way, we have to check `passesOcspPolicy(*selected_ctx)` out of the loop because not passing the policy is a hard fail where it appears like ecdsa is not.

It is probably wrong as it currently is then and `passesOcspPolicy` should be part of that condition?

Reply:
This is a good test, added it.